src/awkward/_v2/contents/unionarray.py

```diff
@@ -712,7 +718,13 @@ def _argsort_next(
     def _sort_next(
         self, negaxis, starts, parents, outlength, ascending, stable, kind, order
     ):
+        if len(self) == 0:
+            return self
+
         simplified = self.simplify_uniontype(mergebool=True)
+        if len(simplified) == 0:
+            return simplified
```

Comment:
Same here.

Reply:
this enable sorting of zero length arrays

https://github.com/scikit-hep/awkward-1.0/pull/1141/files/b1bb82db80d16875bafead8467b6d01d2ae3d3d8#diff-3943a97658fec00f9029d697ac275074416842b955db8f0d66c200ae2f20bc5cL820-L822